_posts/2020-10-27-rails-upgrade-shims.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: post
+title: "Shims that you might encounter during a Rails Upgrade"
+date: 2020-10-27 10:00:00
+categories: ["rails", "upgrades"]
+author: "luciano"
+---
+
+When upgrading a Rails application, you might find that sometimes functionality is extracted from Rails and moved into a new gem. These gems are called [shims](https://medium.com/@ujjawal.dixit/what-is-a-shim-72d9ac5d8620), and basically allow you to keep using an old functionality, once the core API takes that out.
```

Comment:
> and basically allow you

This sound a little off, how about these alternatives?

> and they will basically allow you to ... 
> and basically allows you ... 

Also, one note about this, shims are not only gems. Shims can also be monkeypatches that you develop to make your migration easier. 

For instance, in the Rails 4.2 -> 5.0 migration PR you might want to add a shim that defines a method like this: 

```ruby
class ActiveRecord::Migration
  def [](_)
    ActiveRecord::Migration
  end
```

That way, when you rename the superclass in all your migrations (required in Rails 5.0: https://blog.bigbinary.com/2016/03/01/migrations-are-versioned-in-rails-5.html) the migrations will work with both Rails 4.2 and 5.0 which is useful when you're dual booting. 😄

Reply:
Good point 👍